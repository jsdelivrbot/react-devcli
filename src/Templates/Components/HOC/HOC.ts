export const statelessJS = (): string => {
    return [
        'import React from \'react\';',
        '',
        'export class REACT__COMPONENT__NAME extends React.Component {',
        '   constructor(props) {',
        '       super(props);',
        '   }',
        '',
        '   render() {',
        '       const Component = this.props.component;',
        '',
        '       return (',
        '           <Component {...this.props}></Component>',
        '       );',
        '   }',
        '};',
    ].join('\n');
};

export const statefulJS = (): string => {
    return [
        'import React from \'react\';',
        '',
        'export class REACT__COMPONENT__NAME extends React.Component {',
        '   constructor(props) {',
        '       super(props);',
        '   }',
        '',
        '   render() {',
        '       const Component = this.props.component;',
        '',
        '       return (',
        '           <Component {...this.props}></Component>',
        '       );',
        '   }',
        '};',
    ].join('\n');
};

export const statelessTS = (): string => {
    return [
        'import React from \'react\';',
        '',
        'export class REACT__COMPONENT__NAME extends React.Component {',
        '   constructor(props) {',
        '       super(props);',
        '   }',
        '',
        '   render() {',
        '       const Component = this.props.component;',
        '',
        '       return (',
        '           <Component {...this.props}></Component>',
        '       );',
        '   }',
        '};',
    ].join('\n');
};

export const statefulTS = (): string => {
    return [
        'import React from \'react\';',
        '',
        'export class REACT__COMPONENT__NAME extends React.Component {',
        '   constructor(props) {',
        '       super(props);',
        '   }',
        '',
        '   render() {',
        '       const Component = this.props.component;',
        '',
        '       return (',
        '           <Component {...this.props}></Component>',
        '       );',
        '   }',
        '};',
    ].join('\n');
};

