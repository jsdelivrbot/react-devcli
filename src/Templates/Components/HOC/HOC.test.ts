import {statefulJS, statelessJS, statefulTS, statelessTS} from './HOC';

describe('HOC template test suite', (): void => {
    test('should match stateful HOC template snapshot for JS language', (): void => {
        expect(statefulJS()).toMatchSnapshot();
    });

    test('should match stateless HOC template snapshot for JS language', (): void => {
        expect(statelessJS()).toMatchSnapshot();
    });

    test('should match stateful HOC template snapshot for TS language', (): void => {
        expect(statefulTS()).toMatchSnapshot();
    });

    test('should match stateless HOC template snapshot for TS language', (): void => {
        expect(statelessTS()).toMatchSnapshot();
    });
});
