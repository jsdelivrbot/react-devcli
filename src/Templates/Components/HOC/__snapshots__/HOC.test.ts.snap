// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC template test suite should match stateful HOC template snapshot for JS language 1`] = `
"import React from 'react';

export class REACT__COMPONENT__NAME extends React.Component {
   constructor(props) {
       super(props);
   }

   render() {
       const Component = this.props.component;

       return (
           <Component {...this.props}></Component>
       );
   }
};"
`;

exports[`HOC template test suite should match stateful HOC template snapshot for TS language 1`] = `
"import React from 'react';

export class REACT__COMPONENT__NAME extends React.Component {
   constructor(props) {
       super(props);
   }

   render() {
       const Component = this.props.component;

       return (
           <Component {...this.props}></Component>
       );
   }
};"
`;

exports[`HOC template test suite should match stateless HOC template snapshot for JS language 1`] = `
"import React from 'react';

export class REACT__COMPONENT__NAME extends React.Component {
   constructor(props) {
       super(props);
   }

   render() {
       const Component = this.props.component;

       return (
           <Component {...this.props}></Component>
       );
   }
};"
`;

exports[`HOC template test suite should match stateless HOC template snapshot for TS language 1`] = `
"import React from 'react';

export class REACT__COMPONENT__NAME extends React.Component {
   constructor(props) {
       super(props);
   }

   render() {
       const Component = this.props.component;

       return (
           <Component {...this.props}></Component>
       );
   }
};"
`;
